openwisp_monitoring/device/api/views.py

```diff
@@ -203,6 +205,64 @@ def _write(self, request, pk):
                 metric.write(client['mac'])
             if created:
                 self._create_clients_graph(metric)
+        if 'resources' not in data:
+            return
+        if 'load' in data['resources']:
```

Comment:
ok, this method is becoming huge, let's improve this part (later on we'll likely improve the part above too).

I would do something like the following:

```python
if 'resources' not in data:
    return
self._write_cpu(data)
self._write_disk(data)
self._write_memory(data)
```

and move the relevant logic to each of those methods.